data/soap/src/main/java/org/teiid/spring/data/soap/SoapConnectionFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.teiid.spring.data.soap;
+
+import java.io.IOException;
+
+import javax.security.auth.Subject;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.teiid.spring.data.BaseConnectionFactory;
+import org.teiid.translator.TranslatorException;
+import org.teiid.translator.ws.WSConnection;
+import org.teiid.ws.cxf.BaseWSConnection;
+import org.teiid.ws.cxf.WSConnectionFactory;
+
+@ConfigurationProperties(prefix="spring.teiid.soap")
+public class SoapConnectionFactory extends BaseConnectionFactory<WSConnection> {
+
+    private WSConnectionFactory wsConnectionFactory;
+
+    public SoapConnectionFactory(SoapConfiguration config) throws TranslatorException {
+        super("soap", "spring.teiid.soap");
+        this.wsConnectionFactory = new WSConnectionFactory(config);
+    }
+
+    @Override
+    public WSConnection getConnection() throws Exception {
+        //none of the security methods are yet supported
```

Comment:
This was one of my main reasons I did not bring CXF for REST, in Spring world it is always uphill battle to make non-spring stuff to work. Can we get maybe  simple username/password work?

Reply:
It does work with the null subject change.